src/app/shared/header/case-header.component.spec.ts

```diff
@@ -39,25 +30,39 @@ describe('CaseHeaderComponent', () => {
       .configureTestingModule({
         declarations: [
           CaseHeaderComponent,
-          CaseReferencePipe
+          CaseReferencePipe,
+          LabelSubstitutorDirective,
+          // Mock
+          MarkdownComponent
         ],
-        providers: []
+        providers: [
+          FieldsUtils,
+          LabelSubstitutionService,
+        ]
       })
       .compileComponents();
 
     fixture = TestBed.createComponent(CaseHeaderComponent);
     component = fixture.componentInstance;
-
     component.caseDetails = CASE_DETAILS;
 
     de = fixture.debugElement;
     fixture.detectChanges();
   }));
 
-  it('should render a header based on the case details', () => {
+  it('should render a header with case reference when title display is empty', () => {
     let header = de.query($HEADING);
     expect(header).toBeTruthy();
     expect(text(header)).toEqual('#1234-5678-9012-3456');
   });
 
+  it('should render a header with markdown element when title display is not empty', () => {
+    component.caseDetails.state.title_display = 'Title';
+    component.ngOnInit();
+    fixture.detectChanges();
+
+    let header = de.query($MARKDOWN);
+    expect(header).toBeTruthy();
```

Comment:
Could we make it so that the actual content of the title, in this case "Title", is tested as well?

Reply:
Done